client/gutenberg/editor/index.js

```diff
@@ -1,11 +1,28 @@
 import page from 'page';
-import { makeLayout, render as clientRender } from 'calypso/controller';
+import { makeLayout, redirectLoggedOut, render as clientRender } from 'calypso/controller';
+import { getSiteFragment } from 'calypso/lib/route';
 import { siteSelection, sites } from 'calypso/my-sites/controller';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getImmediateLoginLocale } from 'calypso/state/immediate-login/selectors';
 import { authenticate, post, redirect, siteEditor, exitPost } from './controller';
 
+function redirectToPermalinkIfLoggedOut( context, next ) {
+	if ( isUserLoggedIn( context.store.getState() ) ) {
+		return next();
+	}
+	const state = context.store.getState();
+	const siteFragment = context.params.site || getSiteFragment( context.path );
+	const login_locale = getImmediateLoginLocale( state );
+
+	// force full page reload to avoid SSR hydration issues.
```

Comment:
This comment seems to be off; we don't really reload Calypso here, we actually redirect to an endpoint that will redirect us somewhere itself.

Reply:
I updated the comment